### PR TITLE
[Docs] Add two footnotes about PEP 660 support

### PR DESCRIPTION
**Pull request by kngwyu**
It looks like that[PEP 660](https://peps.python.org/pep-0660/)  is implemented in pip 21.3 (see https://pip.pypa.io/en/stable/news/#v21-3).
Or isn't it still recommended to do `pip install -e` without `setup.py`? 🤔

### CODE REVIEW

**Comment by abravalheri** [comment]
Hi @kngwyu thank you very much for the PR.

There is one situation right now that people might still need `setup.py` regardless of the version of `pip`: if they are only using `pyproject.toml` and don't have `setup.cfg`.

Do you think the note still makes sense considering this information?

**Comment by kngwyu** [comment]
> There is one situation right now that people might still need setup.py regardless of the version of pip: if they are only using pyproject.toml and don't have setup.cfg.

Ah thank you for letting me know that! I think that I was confused because I had `setup.cfg` for flake8 so the editable install worked for me without `setup.py`. 

> Do you think the note still makes sense considering this information?

No. I feel like adding a footnote is better than revising this sentence then. Writing everything precisely could make the document too heavy. I'll revise this PR next monday.

**Comment by abravalheri** [comment]
Thank you very much @kngwyu for having a look on this.

**Comment by kngwyu** [comment]
@abravalheri So because of https://github.com/pypa/pip/pull/10577, even an empty `setup.cfg` and `pyproject.toml` would work for editable install, but we should recommend users to use `setup.py`, right?

**Comment by kngwyu** [comment]
And we don't need `setup.py/cfg` with setuptools > 63.0 since it now has PEP 660 backend?
Sorry for many questions 😅

**Comment by kngwyu** [comment]
Thank you for answering!
I revised this PR to only add footnotes.

File: docs/userguide/pyproject_config.rst [hunk 1]
```diff
@@ -211,6 +211,14 @@ however please keep in mind that all non-comment lines must conform with :pep:`5
 
 .. rubric:: Notes
 
+.. [#pep660-status] Editable install without ``setup.py`` will be supported from
+   ``setuptools >= 63.0.0``, but this feature will remain experimental for a while.
+   Check https://github.com/pypa/setuptools/issues/2816 for detail.
```
Comment (MuellerSeb) [inline docs/userguide/pyproject_config.rst:216]: Since setuptools 63 was released without this support, I am a bit confused now.

Comment (abravalheri) [reply docs/userguide/pyproject_config.rst:216]: Sorry for the confusion. There was a different major change that got in the pipeline before this one.

PEP 660 will probably get in on v64. It is currently under tests.

Comment (kngwyu) [reply docs/userguide/pyproject_config.rst:216]: Ah, thank you for pointing it out. I misunderstood the situation.

**Comment by kngwyu** [comment]
> Would you mind if we merge this PR together with the feature/pep660 branch? This way it lands on the docs together with the actual implementation.

Sounds good to me!

File: docs/userguide/pyproject_config.rst [hunk 2]
```diff
@@ -7,8 +7,8 @@ Configuring setuptools using ``pyproject.toml`` files
 .. note:: New in 61.0.0
 
 .. important::
-   For the time being, ``pip`` still might require a ``setup.py`` file
-   to support :doc:`editable installs <pip:cli/pip_install>`.
+   For the time being [#pep660-status]_, ``pip`` still might require a ``setup.py`` file
+   to support :doc:`editable installs <pip:cli/pip_install>` [#setupcfg-caveats]_.
```
Comment (pradyunsg) [inline docs/userguide/pyproject_config.rst:11]: It does not -- and this note can be removed.

`pip` will gracefully figure out whether to use PEP 660, or invoke setup.py manually. I believe this was from a pre-PEP 660 implementation days.

**Comment by abravalheri** [comment]
@kngwyu #3488 is just around the corner so I decided to "clean the PR queue".
I changed a little bit the notes to avoid mentioning a specific version number, since we never know if the version gets bumped meanwhile...

Thank you very much for the contribution.

**Pull request merged**